integration-provisioning/src/test/java/org/symphonyoss/integration/provisioning/service/CompanyCertificateServiceTest.java

```diff
@@ -155,15 +164,46 @@ public void testGetEmptyCommonNameFromApplicationCertificate() {
     Keystore keystore = new Keystore();
     keystore.setPassword(DEFAULT_KEYSTORE_PASSWORD);
 
-    Application application2 = new Application();
-    application2.setId(APP_ID);
-    application2.setComponent(APP_TYPE);
-    application2.setKeystore(keystore);
+    Application application = new Application();
+    application.setId(APP_ID);
+    application.setComponent(APP_TYPE);
+    application.setKeystore(keystore);
 
-    String name = service.getCommonNameFromApplicationCertificate(application2);
+    String name = service.getCommonNameFromApplicationCertificate(application);
     assertTrue(StringUtils.isEmpty(name));
   }
 
+  @Test(expected = LoadKeyStoreException.class)
+  public void testFailGetCommonNameFromApplicationCertificate()
+      throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
+    Keystore keystore = new Keystore();
+    keystore.setPassword(INVALID_KEYSTORE_PASSWORD);
+    keystore.setFile(MOCK_KEYSTORE_FILE);
+
+    Application application = new Application();
+    application.setId(APP_ID);
+    application.setComponent(APP_TYPE);
+    application.setKeystore(keystore);
+
+    service.getCommonNameFromApplicationCertificate(application);
+  }
+
+  @Test
+  public void testGetCommonNameFromApplicationCertificateEmptyAliases()
+      throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
+    Keystore keystore = new Keystore();
```

Comment:
The creation of these keystores are different, they use different parameters. So I think it's better to let it as it's now.